### PR TITLE
Copy *tune_gemm* from `triton-mlir` branch to `main_perf` branch

### PR DESCRIPTION
**Pull request by brunomazzottiamd**
The source commit in `triton-mlir` branch is the following one:
```
commit cf44637139ba441342b909977f51f7cfec2c9963
Author: Lixun Zhang <Lixun.Zhang@amd.com>
Date:   Tue Jul 23 14:22:01 2024 -0500

    [tuning] gemm tuning script v3.3 (#606)
```

*tune_gemm* was copied from the source branch directory `scripts/amd/gemm`
to the destination branch directory `python/perf-kernels/tune_gemm`.

The SHA-256 hashes of *tune_gemm* files are the following ones:
```
423aef1deb6c60f6578a1ecfc94d2473f8746b00d0368c553d31641fcfa5e354  README.md
46ab93978fee33f75df23332f12546dae7910478c391f08b7b1ebd415d8266b7  icache_flush.py
f18711544641b810a652e6a6629bfa2b613f6ade87399e88fdf05b81d4af58a4  matmul.py
84a1c80ede36d3154e51188276eda2d2d0f52ed4f496ff69349c390d83b8ec10  matmul_kernel.py
2812b40183637bc8d7e47d283c7d66b1792134a43de76f3eacf7b9b3e1c2431a  one_config.py
0ac09c33b0173cea06ddabbf9f4e3afa1816781dea4fdcce5894a7e7d6a80e19  rocprof_gemm.py
00eff41cf1c0bfc41d623e42b51706af67639fec76146741e2067d2a93e0148a  utils/file_generator.py
cb7afb773ccee835b00396cccf87e0d44fe513131161f031fae42453725b3c82  utils/utils.py
59f23811b660e49e566927853926a21f02a7014bb19c8ea67e6b382db6c59900  tune_gemm.py
e787f35d750b869f113b3c01692f64243a9cb8a71a18ade2f0465f614f7284e4  tune_gemm.sh
```

The files were kept as-is despite `pre-commit` intentions to change them.

### CODE REVIEW

**Comment by vgokhale** [comment]
Thanks Bruno. Can you please move them to a new folder called perf-kernels/tune_gemm?

**Comment by brunomazzottiamd** [comment]
> Thanks Bruno. Can you please move them to a new folder called perf-kernels/tune_gemm?

Of course, as you wish.

**Comment by brunomazzottiamd** [comment]
Vinayak, do you mean a new `${ROOT}/perf-kernels/tune_gemm` folder or under the existing folder `perf-kernels` folder, i.e. `${ROOT}/python/perf-kernels/tune_gemm`?

**Comment by vgokhale** [comment]
> Vinayak, do you mean a new ${ROOT}/perf-kernels/tune_gemm folder or under the existing folder perf-kernels folder, i.e. ${ROOT}/python/perf-kernels/tune_gemm?

main_perf branch does not allow file changes outside of triton/python/perf-kernels. That is one of the reasons why your tests are failing right now. So to answer your question, ${ROOT}/python/perf-kernels/tune_gemm.

**Comment by brunomazzottiamd** [comment]
> Thanks Bruno. Can you please move them to a new folder called perf-kernels/tune_gemm?

Now it's done.

**Comment by brunomazzottiamd** [comment]
Source branch was updated to reflect latest *tune_gemm* additions made by Lixun.

**Comment by zhanglx13** [comment]
@vgokhale @brunomazzottiamd 
Should we land this PR asap so that I can rebase/migrate my [latest changes](https://github.com/ROCm/triton/pull/616) to the tuning script onto the main_perf branch?
Or should we wait for my changes to land first? It's gonna take 1 or 2 days.

File: python/perf-kernels/tune_gemm/matmul.py
Comment (xiaohuguo2023) [inline python/perf-kernels/tune_gemm/matmul.py:1]: not sure if this script is still being used ?

Comment (brunomazzottiamd) [reply python/perf-kernels/tune_gemm/matmul.py:1]: Xiaohu, I really don't know if `matmul.py` is still being used. My intention was to just move the entire *tune_gemm* directory as-is to `main_perf` branch.

What do you think about keeping the PR as-is and evaluating `matmul.py` usefulness in another scope? We can open a cleanup PR later if it's not being used anymore.

Thanks for your comment.

**Comment by brunomazzottiamd** [comment]
> @vgokhale @brunomazzottiamd Should we land this PR asap so that I can rebase/migrate my [latest changes](https://github.com/ROCm/triton/pull/616) to the tuning script onto the main_perf branch? Or should we wait for my changes to land first? It's gonna take 1 or 2 days.

As far as I can see, Xiaohu's changes are easy to incorporate later. I can redo my PR if it's better for the team. For me it's just a matter of prioritizing the PRs. @vgokhale, your opinion is valuable here.

File: python/perf-kernels/tune_gemm/matmul.py
Comment (zhanglx13) [reply python/perf-kernels/tune_gemm/matmul.py:1]: matmul.py, rocprof_gemm.py and tune_gemm.sh were added by Shucai as the first version of the gemm tuning script: https://github.com/ROCm/triton/pull/309
I agree we can clean things up later and check with Shucai to make sure these files are not used anymore.

Comment (brunomazzottiamd) [reply python/perf-kernels/tune_gemm/matmul.py:1]: Let's open a cleanup ticket describing the candidates for removal.

**Comment by brunomazzottiamd** [comment]
> @vgokhale @brunomazzottiamd Should we land this PR asap so that I can rebase/migrate my [latest changes](https://github.com/ROCm/triton/pull/616) to the tuning script onto the main_perf branch? Or should we wait for my changes to land first? It's gonna take 1 or 2 days.

@zhanglx13, go ahead and merge your latest changes! I'll redo this PR later. Thanks for you patience.

**Comment by brunomazzottiamd** [comment]
@zhanglx13 said his changes (https://github.com/ROCm/triton/pull/616) aren't that important and we can merge this PR first.

**Review by vgokhale — APPROVED** [review]

**Comment by micmelesse** [comment]
@brunomazzottiamd  please don't merge pr if one of the check is failing

**Comment by brunomazzottiamd** [comment]
> @brunomazzottiamd please don't merge pr if one of the checks is failing. The code formating check is failing in this case. Can you do a follow up pr to fix it

Hi @micmelesse. I didn't notice the check failed, sorry for that...

However, I consciously by-passed the code formatting pre-commit hook with `git commit --no-verify`. My idea was to copy *tune_gemm* script AS IS from `triton-mlir` branch, without changing anything that was not necessary for the correct operation. So we could minimize code review overhead to people already used to *tune_gemm* code.

I can apply Triton's code formatting pre-commit hook in another PR if it's the right thing to do, but code in `triton-mlir` branch wasn't following this policy.

**Comment by brunomazzottiamd** [comment]
By the way, can we disable the merge button if checks are failing?

**Comment by micmelesse** [comment]
yep, the merge button is disabled now if the checks don't pass.

File: python/perf-kernels/tune_gemm/matmul.py
Comment (brunomazzottiamd) [reply python/perf-kernels/tune_gemm/matmul.py:1]: Being addressed by https://github.com/ROCm/triton/pull/629.

Comment (brunomazzottiamd) [reply python/perf-kernels/tune_gemm/matmul.py:1]: Resolved by #629.

**Pull request merged**
